### PR TITLE
db/kpi-plugin

### PR DESCRIPTION
**Pull request by karolkolodziej**
Added `KPIComponent` to `HighchartsPlugin`.

- [x] TS

### CODE REVIEW

**Review by sebastianbochan — APPROVED** [review]

File: ts/Dashboard/Actions/Bindings.ts [hunk 1]
```diff
@@ -63,23 +63,23 @@ class Bindings {
         options: Bindings.ComponentOptions,
         cell?: Cell
     ): ComponentTypes | undefined {
-        const compontentContainer = document.getElementById(options.cell);
+        const componentContainer = document.getElementById(options.cell);
```
Comment (pawellysy) [inline ts/Dashboard/Actions/Bindings.ts:66]: Add the type narrowing here as `HTMLComponent | null` and delete it elsewhere

File: ts/Dashboard/Actions/Bindings.ts [hunk 2]
```diff
@@ -63,23 +63,23 @@ class Bindings {
...
+        if (componentContainer) {
             const ComponentClass = Component.getComponent(options.type);
 
             switch (options.type) {
                 case 'html':
                     component = new HTMLComponent(merge(
                         options,
                         {
-                            parentElement: compontentContainer as HTMLDOMElement,
+                            parentElement: componentContainer as HTMLDOMElement,
```
Comment (pawellysy) [inline ts/Dashboard/Actions/Bindings.ts:82]: Delete type casting

File: ts/Dashboard/Actions/Bindings.ts [hunk 3]
```diff
@@ -89,7 +89,7 @@ class Bindings {
                         component = new ComponentClass(merge(
                             options,
                             {
-                                parentElement: compontentContainer as HTMLDOMElement,
+                                parentElement: componentContainer as HTMLDOMElement,
```
Comment (pawellysy) [inline ts/Dashboard/Actions/Bindings.ts:92]: delete type casting

File: ts/Dashboard/Actions/Bindings.ts [hunk 4]
```diff
@@ -101,18 +101,20 @@ class Bindings {
                         component = new ComponentClass(merge(
                             options,
                             {
-                                parentElement: compontentContainer as HTMLDOMElement
+                                parentElement: componentContainer as HTMLDOMElement
```
Comment (pawellysy) [inline ts/Dashboard/Actions/Bindings.ts:104]: delete type casting

File: ts/Dashboard/Actions/Bindings.ts [hunk 5]
```diff
@@ -101,18 +101,20 @@ class Bindings {
...
-                        {
-                            parentElement: compontentContainer as HTMLDOMElement
-                        })
-                    )
+                case 'KPI':
+                    if (ComponentClass) {
+                        component = new ComponentClass(merge(
+                            options,
+                            {
+                                parentElement: componentContainer as HTMLDOMElement
```
Comment (pawellysy) [inline ts/Dashboard/Actions/Bindings.ts:114]: delete type casting

**Review by pawellysy — CHANGES_REQUESTED** [review]
Good job! Minor changes requested tho

**Review by bre1470 — COMMENTED** [review]
It looks a bit strange with the default chart height of 400 pixel. I would expect that a KPI by default becomes a squared component.

**Comment by karolkolodziej** [comment]
@bre1470 I agree but this was not a scope of this issue. I will create a separate one where we can discuss all the default options and styling for KPI.

**Review by bre1470 — APPROVED** [review]

**Review by pawellysy — APPROVED** [review]
LGTM!

**Pull request merged**